frontend/src/components/smart-contracts/ApproveERC20.tsx

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import {
+  useSmartContracts,
+  SmartContractActions,
+  SmartContractAddresses,
+} from 'components/smart-contracts/useSmartContracts';
+import { TransactionModal } from 'components/smart-contracts/TransactionModal';
+import { ethers, constants } from 'ethers';
+
+export const ApproveERC20 = ({
```

Comment:
Same as another comment - would like to see as much logic as possible done a level up or in the provider, and then a `<Modal>` just uses `const { approved } = useContracts()` to display. Let me know what you think!